Update telling-git-about-your-signing-key.md

MichaelDeBoey opened the pull request:
### Why:

macOS now comes with zsh instead of bash installed

### What's being changed:

<!-- Share artifacts of the changes, be they code snippets, GIFs or screenshots; whatever shares the most context. -->

### Check off the following:
- [x] I have reviewed my changes in staging. (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**)
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

MichaelDeBoey committed:
Update telling-git-about-your-signing-key.md

janiceilene commented:
@MichaelDeBoey Thanks so much for opening a PR! I'll get this triaged for review ✨

ajs256 commented on content/github/authenticating-to-github/telling-git-about-your-signing-key.md:
````diff
@@ -30,12 +30,12 @@ If you have multiple GPG keys, you need to tell Git which one to use.
 1. If you aren't using the GPG suite, paste the text below to add the GPG key to your bash profile:
   ```shell
-  $ test -r ~/.bash_profile && echo 'export GPG_TTY=$(tty)' >> ~/.bash_profile
+  $ test -r ~/.zshrc && echo 'export GPG_TTY=$(tty)' >> ~/.zshrc
   $ echo 'export GPG_TTY=$(tty)' >> ~/.profile
````
```suggestion
  test -r ~/.zshrc && echo 'export GPG_TTY=$(tty)' >> ~/.zshrc
  echo 'export GPG_TTY=$(tty)' >> ~/.profile
```

According to [the style guide](https://github.com/github/docs/blob/main/contributing/content-style-guide.md#code-blocks), you should only include the `$` when you are showing the command's output in the same block. 

> Disclaimer: **I am not a GitHub employee.**

MichaelDeBoey replied:
I left it because it was already in place, but I'm happy to remove them if necessary

MichaelDeBoey committed:
Update content/github/authenticating-to-github/telling-git-about-your-signing-key.md

Co-authored-by: hubwriter <hubwriter@github.com>

MichaelDeBoey committed:
Update content/github/authenticating-to-github/telling-git-about-your-signing-key.md

Co-authored-by: hubwriter <hubwriter@github.com>

hubwriter commented:
@MichaelDeBoey - Many thanks for raising this PR. 👍 

I think there are a couple of things that we may need to clarify in these instructions, so I'll get someone who knows more about the z shell internally here at GitHub to take a look at this.

hubwriter committed:
Give alternatives for Mac using zsh and bash

I've also modified the commands so that .zprofile (or .profile) is only modified if .zshrc (or .bash_profile) doesn't exist.

hubwriter commented:
@MichaelDeBoey - After taking advice on this, I've made some further changes to provide alternative zsh and bash commands for Mac users - see my changes here:
https://github.com/github/docs/pull/4609/commits/200d0ff7ac6dbf87af4b4b5882e2f2441960803f

I've also modified the commands so that `.zprofile` (or `.profile`) is only modified if `.zshrc` (or `.bash_profile`) doesn't exist.

hubwriter commented:
Colleague review (by DM):

> @hubwriter the only issue I see is that the // look like they should be line continuation characters \?

hubwriter commented on content/github/authenticating-to-github/telling-git-about-your-signing-key.md:
````diff
@@ -28,16 +28,16 @@ If you have multiple GPG keys, you need to tell Git which one to use.
   ```shell
-  $ test -r ~/.bash_profile && echo 'export GPG_TTY=$(tty)' >> ~/.bash_profile
-  $ echo 'export GPG_TTY=$(tty)' >> ~/.profile
+  $ if [ -r ~/.zshrc ]; then echo 'export GPG_TTY=$(tty)' >> ~/.zshrc; //
````
```suggestion
  $ if [ -r ~/.zshrc ]; then echo 'export GPG_TTY=$(tty)' >> ~/.zshrc; \
```

hubwriter commented on content/github/authenticating-to-github/telling-git-about-your-signing-key.md:
````diff
@@ -28,16 +28,16 @@ If you have multiple GPG keys, you need to tell Git which one to use.
+  ```
+  Alternatively, if you use the `bash` shell, run this command:
+  ```shell
+  $ if [ -r ~/.bash_profile ]; then echo 'export GPG_TTY=$(tty)' >> ~/.bash_profile; //
````
```suggestion
  $ if [ -r ~/.bash_profile ]; then echo 'export GPG_TTY=$(tty)' >> ~/.bash_profile; \
```

hubwriter commented on content/github/authenticating-to-github/telling-git-about-your-signing-key.md:
````diff
@@ -90,10 +90,10 @@ If you have multiple GPG keys, you need to tell Git which one to use.
   ```shell
-  $ test -r ~/.bash_profile && echo 'export GPG_TTY=$(tty)' >> ~/.bash_profile
-  $ echo 'export GPG_TTY=$(tty)' >> ~/.profile
+  $ if [ -r ~/.bash_profile ]; then echo 'export GPG_TTY=$(tty)' >> ~/.bash_profile; //
````
```suggestion
  $ if [ -r ~/.bash_profile ]; then echo 'export GPG_TTY=$(tty)' >> ~/.bash_profile; \
```

hubwriter committed:
Update content/github/authenticating-to-github/telling-git-about-your-signing-key.md

hubwriter committed:
Update content/github/authenticating-to-github/telling-git-about-your-signing-key.md

hubwriter committed:
Update content/github/authenticating-to-github/telling-git-about-your-signing-key.md

hubwriter committed:
Merge branch 'main' into patch-1

hubwriter reviewed (approved):
@MichaelDeBoey - Thanks again for raising this PR. 
Merging it now.

hubwriter committed:
Merge branch 'main' into patch-1
